sktime/forecasting/hf_patchtst_forecaster.py

```diff
@@ -0,0 +1,548 @@
+        self.config = config
+        self.training_args = training_args
+        self.compute_metrics = compute_metrics
+        self.callbacks = callbacks
+
+        self._config = deepcopy(self.config)
+        super().__init__()
+        if self.fit_strategy not in ["full", "minimal", "zero-shot"]:
+            raise ValueError("unexpected fit_strategy passed in argument")
+
+    def _fit(self, y, fh, X=None):
+        """Fits the model.
+
+        Parameters
+        ----------
+        y : pandas DataFrame
+            pandas dataframe containing single or multivariate data
+
+        fh : Forecasting Horizon object
+            used to determine forecasting horizon for predictions
+
+        Returns
+        -------
+        self : a reference to the object
+        """
+        self._y = y
+        self.fh_ = int(max(fh.to_relative(self.cutoff)))
+        self.y_columns = y.columns
+        # if no model_path was given, initialize new full model from config
+        if not self.model_path:
+            self._config["num_input_channels"] = len(self.y_columns)
+            if "prediction_length" in self._config.keys():
+                if self._config["prediction_length"] > self.fh_:
+                    warnings.warn(
+                        "Found prediction length inside config larger"
+                        "than passed fh, will use larger of the two"
+                    )
+                elif self._config["prediction_length"] <= self.fh_:
+                    warnings.warn(
+                        "Found fh length inside config larger"
+                        "than passed prediction_length, will use larger of the two"
+                    )
+                    self._config["prediction_length"] = int(self.fh_)
+            else:
+                self._config["prediction_length"] = int(self.fh_)
+            context_length = self._config["context_length"]
+            del self._config["context_length"]
+            config = PatchTSTConfig(
+                context_length=context_length,
+                **self._config,
+            )
+            self.model = PatchTSTForPrediction(config)
+        else:
+            # model_path was given, initialize with model_path
+            self.model = PatchTSTForPrediction.from_pretrained(self.model_path)
+            if not isinstance(self.model.model, PatchTSTModel):
+                raise ValueError(
+                    "This estimator requires a `PatchTSTModel`, but "
+                    f"found {self.model.model.__class__.__name__}"
+                )
+
+        if self.fit_strategy != "zero-shot":
+            # initialize dataset
+            y_train, y_test = temporal_train_test_split(
+                y, train_size=1 - self.validation_split, test_size=self.validation_split
+            )
+            train_dataset = PyTorchDataset(
+                y_train,
+                context_length=self.model.config.context_length,
+                prediction_length=self.fh_,
```

Comment:
use `self.model.config.prediction_length` here instead as `self.fh_` is not updated if `self.model.config.prediction_length` is greater than `fh`, same goes for eval_dataset